src/I18n/I18n.php

```diff
@@ -125,16 +126,71 @@ public static function translators()
     public static function translator($name = 'default', $locale = null, callable $loader = null)
     {
         if ($loader !== null) {
-            $locale = $locale ?: static::locale();
-
-            $loader = static::translators()->setLoaderFallback($name, $loader);
-
-            $packages = static::translators()->getPackages();
-            $packages->set($name, $locale, $loader);
+            static::setTranslator($name, $locale, $loader);
 
             return null;
         }
 
+        return self::getTranslator($name, $locale);
+    }
+
+    /**
+     * Configures future translators, this is achieved by passing a callable
+     * as the last argument of this function.
+     *
+     * ### Example:
+     *
+     * ```
+     *  I18n::translator('default', 'fr_FR', function () {
+     *      $package = new \Aura\Intl\Package();
+     *      $package->setMessages([
+     *          'Cake' => 'Gâteau'
+     *      ]);
+     *      return $package;
+     *  });
+     *
+     *  $translator = I18n::translator('default', 'fr_FR');
+     *  echo $translator->translate('Cake');
+     * ```
+     *
+     * You can also use the `Cake\I18n\MessagesFileLoader` class to load a specific
+     * file from a folder. For example for loading a `my_translations.po` file from
+     * the `src/Locale/custom` folder, you would do:
+     *
+     * ```
+     * I18n::translator(
+     *  'default',
+     *  'fr_FR',
+     *  new MessagesFileLoader('my_translations', 'custom', 'po');
+     * );
+     * ```
+     *
+     * @param string $name The domain of the translation messages.
+     * @param string|null $locale The locale for the translator.
+     * @param callable|null $loader A callback function or callable class responsible for
+     * constructing a translations package instance.
+     * @return void
+     */
+    public static function setTranslator($name = 'default', $locale = null, callable $loader)
```

Comment:
The problem is it seems that the default name of value `default` is expected. I was looking into changing it to `($name, callable $loader, $locale = null)`.

If we're OK with that I'll do it. BC can be maintained by translator() by passing `default` as it is.